full-service/src/service/transaction_builder.rs

```diff
@@ -249,7 +249,87 @@ impl<FPR: FogPubkeyResolver + 'static> WalletTransactionBuilder<FPR> {
 
         let view_account_key = account.view_account_key()?;
         let view_private_key = account.view_private_key()?;
-        let reserved_subaddresses = ReservedSubaddresses::from(&view_account_key);
+        let reserved_subaddresses = match account.account_key() {
+            Ok(account_key) => ReservedSubaddresses::from(&account_key),
+            Err(_) if !account.fog_enabled => ReservedSubaddresses::from(&view_account_key),
+            Err(_) => {
```

Comment:
This is the inelegant part. If we want to clean this up, we can start storing the fog parameters inside the `Account` row, but that would require a migration, so maybe this is acceptable.
I haven't tested this with a hardware wallet - I am not sure what the flow for that is at the moment, but I did make sure this code appears to be working by commenting out the `match` statement and just running it on my fog-enabled non-view-only account - so at least in that scenario it works as expected.

Reply:
hardware wallets were borked due `import_view_only_from_hardware_wallet_with_fog` not creating a fog-enabled subaddress for the change; fixed that and pushed the commit to his branch.